### PR TITLE
include/assert: clean up ceph assertion macros

### PR DESCRIPTION
**Pull request by liewegas**

### CODE REVIEW

**Comment by liewegas** [comment]
test this please

File: src/include/assert.h [hunk 1]
```diff
@@ -129,6 +129,13 @@ using namespace ceph;
    ? __CEPH_ASSERT_VOID_CAST (0)					\
    : __ceph_assert_fail (__STRING(expr), __FILE__, __LINE__, __CEPH_ASSERT_FUNCTION))
 
+// this variant will *never* get compiled out to NDEBUG in the future.
+// (ceph_assert currently doesn't either, but in the future it might.)
```
Comment (gregsfortytwo) [inline src/include/assert.h]: I'm not very familiar with how our current asserts work with debug that lets them be disabled or not. Can we do something to make sure the tooling is set up properly right now, which guarantees ceph_assert_always really does get compiled in?

**Comment by gregsfortytwo** [comment]
Basically looks fine.

File: src/include/assert.h (hunk 1 above)
Comment (liewegas) [inline src/include/assert.h]: Our aren't conditional (yet)... It's the lack of an #ifdef NDEBUG where this comment goes that prevents it from compiling out.  And/or the lack of a similar comment says "do not ever compile this out" next to the ceph_assert definition a few lines up.

**Comment by tchaikov** [comment]
tested at 

* http://pulpito.ceph.com/kchai-2016-07-02_20:22:26-rados-wip-kefu-testing---basic-smithi/
* http://pulpito.ceph.com/kchai-2016-07-02_09:03:22-rados-wip-kefu-testing---basic-mira/

**Pull request merged**